### PR TITLE
Add com.jgraph.drawio.desktop

### PR DESCRIPTION
**Pull request by ghisvail**
Flatpak packaging for [draw.io](https://github.com/jgraph/drawio-desktop).

### CODE REVIEW

**Comment by ghisvail** [comment]
screenshot from 2018-04-30 09-12-21

**Comment by nedrichards** [comment]
bot, build com.jgraph.drawio

**Comment by nedrichards** [comment]
Similar to Zulip, is there a reason why this can't be built from source code? That would possibly let us enable it on more architectures as well as being better and more reproducible practice.

**Comment by ghisvail** [comment]
> is there a reason why this can't be built from source code?

Because it is faster to reach a first iteration that way? I have had little success with building electron projects from source so far (see my past [comment](https://github.com/flathub/flathub/pull/350#issuecomment-384116977)).

I'd be happy to revisit it (myself or with assistance from someone in the community) once a first iteration is out.

**Comment by TingPing** [comment]
> I'd be happy to revisit it (myself or with assistance from someone in the community) once a first iteration is out.

It is very easy for it to just stay in that state forever, especially for non-upstream packagers.

**Comment by nedrichards** [comment]
OK, same as before then ;-)

- Can you open a PR upstream to send the appdata back?
- Can we attach a sourcedump as well, otherwise we're not doing a good job as an open source citizen.

**Comment by ghisvail** [comment]
@nedrichards source added ^.

Quick question. I just noticed upstream is using `com.jgraph.drawio.desktop` as the `appId` in their [electron-builder.json]. Should I use the same for the flatpak files, and if so, is renaming the AppStream metadata and flaptak manifest sufficient?

**Comment by barthalion** [comment]
Preferably yes, files here should follow upstream convention. A rename (and change in the files too) is enough.

**Comment by nedrichards** [comment]
bot, build com.jgraph.drawio

**Comment by nedrichards** [comment]
Yeah - you should absolutely match the upstream appid and we need to do that before shipping, otherwise things get awkward.

**Comment by ghisvail** [comment]
@barthalion, @nedrichards could you guys let me know whether this is enough? ^

File: com.jgraph.drawio.desktop.json [hunk 1]
```diff
@@ -30,7 +30,7 @@
         "install -Dm644 usr/share/icons/hicolor/0x0/apps/draw.io.png /app/share/icons/hicolor/512x512/apps/com.jgraph.drawio.png",
```
Comment (nedrichards) [inline com.jgraph.drawio.desktop.json]: the icons also need to be exported with the new name. all files that are exported need the app-id

File: com.jgraph.drawio.desktop.json [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    {
+      "name": "draw.io",
+      "buildsystem": "simple",
+      "build-commands": [
+        "ar -x draw.io-*.deb",
+        "rm -f draw.io-*.deb",
+        "tar -xf data.tar.xz",
+        "cp -r opt/* /app",
+        "install -Dm755 draw.io.sh /app/bin/draw.io",
+        "install -Dm644 usr/share/applications/draw.io.desktop /app/share/applications/com.jgraph.drawio.desktop",
```
Comment (nedrichards) [inline com.jgraph.drawio.desktop.json]: I believe that this has to be com.jgraph.drawio.desktop.desktop

**Comment by nedrichards** [comment]
You need to change a few more bits. If you build it locally then you can check the exports directory and make sure everything's correct. The build process will tell you about missing files as well.

File: com.jgraph.drawio.desktop.json (hunk 2 above)
Comment (ghisvail) [reply com.jgraph.drawio.desktop.json]: Done

File: com.jgraph.drawio.desktop.json (hunk 1 above)
Comment (ghisvail) [reply com.jgraph.drawio.desktop.json]: Done

**Comment by ghisvail** [comment]
The build runs fine locally.

**Comment by nedrichards** [comment]
bot, build com.jgraph.drawio.desktop

**Comment by ghisvail** [comment]
@davidjgraph Thanks for the heads-up. Updated ^

**Comment by nedrichards** [comment]
bot, build com.jgraph.drawio.desktop

**Comment by ghisvail** [comment]
@nedrichards  looks like we're good?

**Comment by nedrichards** [comment]
Repository has been created: https://github.com/flathub/com.jgraph.drawio.desktop

**Pull request closed without merging**